symbolic/src/Data/Macaw/Symbolic/CrucGen.hs

```diff
@@ -1755,6 +1756,22 @@ addSwitch blockLabelMap idx possibleAddrs = do
   mapM_ addStmt stmts
   addTermStmt termStmt
 
+
+addRegUpdateForBlock :: (OrdF (M.ArchReg arch)) => MacawSymbolicArchFunctions arch -> M.ArchSegmentOff arch -> CrucGen arch ids s ()
+addRegUpdateForBlock archFns startAddr = do
+  mp <- updatesFromRegStruct (crucGenRegAssignment archFns)
+  void $ evalMacawStmt $ MacawArchStateUpdate (M.segoffAddr startAddr) mp
+ where
+  updatesFromRegStruct ::
+    (OrdF (M.ArchReg arch)) =>
+    Assignment (M.ArchReg arch) ctx ->
+    CrucGen arch ids s (MapF.MapF (M.ArchReg arch) (MacawCrucibleValue (CR.Atom s)))
+  updatesFromRegStruct Empty = pure MapF.empty
+  updatesFromRegStruct (rst :> reg) = do
+    rstOf <- updatesFromRegStruct rst
+    rval <- getRegValue reg
+    pure $ MapF.insert reg (MacawCrucibleValue rval) rstOf
```

Comment:
Looks like perhaps this could be written as `foldlMFC` or similar? Maybe not though, and no biggie either way.